### PR TITLE
(multi) merge upstream to master

### PR DESCRIPTION
**Pull request by altendky**
This consists of:
- An ignore merge of the original upstream commit into master
- A real merge of nearly the latest upstream into master
- An ignore merge of the latest upstream (which was an ignore merge of master to upstream) into master

Hopefully after this future merges of updated upstream will be cleaner.

Draft for:
- [x] Self review to make sure at least \_I\_ think this makes sense.
- [x] Address question about new files.
- [x] Unbreak CI back to its regular brokenness

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -44,6 +44,7 @@ The following modules are supported by PyQt5-stubs:
 # Building upstream stubs
 The Dockerfile is used to build all of the stubs for the upstream PyQt5 modules.
 The Dockerfile consists of multiple build layers:
+layers:
```
Comment (BryceBeagle) [inline README.md]: I think this is an unintentional merge artifact?

Comment (altendky) [reply README.md]:
```suggestion
```
Quite so.

**Comment by stlehmann** [comment]
@altendky This PR still has the state of a draft. Is it still work in progress?

**Comment by BryceBeagle** [comment]
I think you meant to mention @altendky instead of me? 🙂

**Comment by stlehmann** [comment]
> 
> 
> I think you meant to mention @altendky instead of me? 🙂

Ah yes, you are right, thanks :blush:

**Comment by altendky** [comment]
I do try to document the reasons for draft marks in the first comment.  Though this one is a bit generic.  `:]`  Since you merged the first pair of PRs I'll go ahead and look over the next batch from https://github.com/stlehmann/PyQt5-stubs/issues/42#issuecomment-692306986 and get them ready.  Hopefully today.

File: PyQt5-stubs/QtTest.pyi [hunk 2]
```diff
@@ -151,6 +151,10 @@ class QTest(sip.simplewrapper):
     @staticmethod
     def mouseClick(window: QtGui.QWindow, button: QtCore.Qt.MouseButton, modifier: typing.Union[QtCore.Qt.KeyboardModifiers, QtCore.Qt.KeyboardModifier] = ..., pos: QtCore.QPoint = ..., delay: int = ...) -> None: ...
     @typing.overload
+    def keySequence(self, widget: QtWidgets.QWidget, keySequence: typing.Union[QtGui.QKeySequence, QtGui.QKeySequence.StandardKey, str, int]) -> None: ...
+    @typing.overload
+    def keySequence(self, window: QtGui.QWindow, keySequence: typing.Union[QtGui.QKeySequence, QtGui.QKeySequence.StandardKey, str, int]) -> None: ...
+    @typing.overload
```
Comment (altendky) [inline PyQt5-stubs/QtTest.pyi:157]: Interesting.  These have `self` and aren't decorated as `@staticmethod`...  lovely inconsistency.  But I guess this is ok?  And it's 'their' doing anyways so a fix should be separate.

File: PyQt5-stubs/QtWidgets.pyi [hunk 3]
```diff
@@ -3014,11 +3014,11 @@ class QFileDialog(QDialog):
     def supportedSchemes(self) -> typing.List[str]: ...
     def setSupportedSchemes(self, schemes: typing.Iterable[str]) -> None: ...
     @staticmethod
-    def getSaveFileUrl(parent: typing.Optional[QWidget] = ..., caption: str = ..., directory: str = ..., filter: str = ..., initialFilter: str = ..., options: typing.Union['QFileDialog.Options', 'QFileDialog.Option'] = ..., supportedSchemes: typing.Iterable[str] = ...) -> typing.Tuple[QtCore.QUrl, str]: ...
+    def getSaveFileUrl(parent: typing.Optional[QWidget] = ..., caption: str = ..., directory: QtCore.QUrl = ..., filter: str = ..., initialFilter: str = ..., options: typing.Union['QFileDialog.Options', 'QFileDialog.Option'] = ..., supportedSchemes: typing.Iterable[str] = ...) -> typing.Tuple[QtCore.QUrl, str]: ...
```
Comment (altendky) [inline PyQt5-stubs/QtWidgets.pyi:3017]: I'm guessing it really could be either `str` or `QtCore.QUrl`, but this doesn't appear to be any customization we did so I'd let this through for now.

File: PyQt5-stubs/QtWidgets.pyi [hunk 4]
```diff
@@ -3828,7 +3828,7 @@ class QGraphicsDropShadowEffect(QGraphicsEffect):
     colorChanged: QtCore.pyqtSignal
     blurRadiusChanged: QtCore.pyqtSignal
     offsetChanged: QtCore.pyqtSignal
-    def setColor(self, color: typing.Union[QtGui.QColor, QtCore.Qt.GlobalColor, QtGui.QGradient]) -> None: ...
+    def setColor(self, color: typing.Union[QtGui.QColor, QtCore.Qt.GlobalColor]) -> None: ...
```
Comment (altendky) [inline PyQt5-stubs/QtWidgets.pyi:3831]: Seems maybe odd but this isn't anything we customized seemingly.

**Review by altendky — COMMENTED** [review]
This is adding the following new files.  Is this good?  I know there was some list of 'unsupported' stuff, though I think that was mostly platform specific.  The other comments are just observations mostly, so these new files are my only particular uncertainty.

- `QtBluetooth.pyi`
- `QtChart.pyi`
- `QtHelp.pyi`
- `QtNetworkAuth.pyi`

File: PyQt5-stubs/QtWidgets.pyi (hunk 4 above)
Comment (BryceBeagle) [reply PyQt5-stubs/QtWidgets.pyi:3831]: For whatever reason, a lot of `QColor`-taking functions are annotated upstream as also accepting `QGradient`s, but don't work

```pycon
effect = QGraphicsDropShadowEffect()
gradient = QGradient()
effect.setColor(gradient)
```
```pytb
Traceback (most recent call last):
  File "<input>", line 1, in <module>
TypeError: setColor(self, Union[QColor, Qt.GlobalColor, QGradient]): argument 1 has unexpected type 'QGradient'
```

Comment (altendky) [reply PyQt5-stubs/QtWidgets.pyi:3831]: I guess this is them fixing one that we hadn't.

Comment (BryceBeagle) [reply PyQt5-stubs/QtWidgets.pyi:3831]: https://github.com/stlehmann/PyQt5-stubs/pull/29/files#diff-92315caecc7de3d85e401ce18dec0ac4R3798

Strange. I had already fixed this specific one. My PR was merged, but somehow that line didn't make it into master 🤔:

https://github.com/stlehmann/PyQt5-stubs/blob/50f45f509d3a03fddb2704b550561f162058ed0b/PyQt5-stubs/QtWidgets.pyi#L3831

**Comment by altendky** [comment]
Do the new files look appropriate?  If so I'll release for review.

**Comment by altendky** [comment]
I guess it really makes more sense to just say "I'm not too sure about those new files..." and mark it ready for review...

**Comment by BryceBeagle** [comment]
Not sure why those new stubs weren't already merged into master. I don't see any issues with them, but then again I've never used any of those modules.

**Comment by altendky** [comment]
@BryceBeagle, I know I confused things a bit with my paperwork on this PR but did you consider your review to be complete?  Or does this need more looking over.

**Review by BryceBeagle — APPROVED** [review]

**Pull request merged**